// testcafe/lib/proxyless/client/input.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const types_1 = require("../types");
const event_descriptor_1 = __importDefault(require("./event-descriptor"));
class ProxylessInput {
    constructor(dispatchEventFn) {
        this._dispatchEventFn = dispatchEventFn;
    }
    async mouseDown(options) {
        const eventOptions = await event_descriptor_1.default.createMouseEventOptions('mousePressed', options);
        return this._dispatchEventFn.single(types_1.EventType.Mouse, eventOptions);
    }
    async mouseUp(options) {
        const eventOptions = await event_descriptor_1.default.createMouseEventOptions('mouseReleased', options);
        return this._dispatchEventFn.single(types_1.EventType.Mouse, eventOptions);
    }
    keyDown(options) {
        const eventOptions = event_descriptor_1.default.createKeyDownOptions(options);
        return this._dispatchEventFn.single(types_1.EventType.Keyboard, eventOptions);
    }
    keyUp(options) {
        const eventOptions = event_descriptor_1.default.createKeyUpOptions(options);
        return this._dispatchEventFn.single(types_1.EventType.Keyboard, eventOptions);
    }
    executeEventSequence(eventSequence) {
        return this._dispatchEventFn.sequence(eventSequence);
    }
    async createMouseMoveEvent(currPosition) {
        const options = await event_descriptor_1.default.createMouseEventOptions('mouseMoved', {
            options: {
                clientX: currPosition.x,
                clientY: currPosition.y,
                button: 'none',
            },
        });
        return {
            type: types_1.EventType.Mouse,
            options,
        };
    }
}
exports.default = ProxylessInput;
module.exports = exports.default;
