// testcafe/lib/proxyless/client/utils.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.calculateMouseButtonValue = exports.getModifiersBit = exports.getModifiersState = exports.calculateKeyModifiersValue = void 0;
const types_1 = require("./types");
// @ts-ignore
const hammerhead_1 = require("../../client/core/deps/hammerhead");
const EMPTY_MODIFIERS = {
    ctrl: false,
    alt: false,
    shift: false,
    meta: false,
};
function calculateKeyModifiersValue(modifiers) {
    let result = 0;
    if (!modifiers)
        return result;
    if (modifiers.ctrl)
        result |= types_1.KeyModifierValues.ctrl;
    if (modifiers.alt)
        result |= types_1.KeyModifierValues.alt;
    if (modifiers.shift)
        result |= types_1.KeyModifierValues.shift;
    if (modifiers.meta)
        result |= types_1.KeyModifierValues.meta;
    return result;
}
exports.calculateKeyModifiersValue = calculateKeyModifiersValue;
function getModifiersState(modifiersBit) {
    const modifiers = hammerhead_1.utils.extend({}, EMPTY_MODIFIERS);
    if (!modifiersBit)
        return modifiers;
    if (modifiersBit & types_1.KeyModifierValues.ctrl)
        modifiers.ctrl = true;
    if (modifiersBit & types_1.KeyModifierValues.alt)
        modifiers.alt = true;
    if (modifiersBit & types_1.KeyModifierValues.shift)
        modifiers.shift = true;
    if (modifiersBit & types_1.KeyModifierValues.meta)
        modifiers.meta = true;
    return modifiers;
}
exports.getModifiersState = getModifiersState;
function getModifiersBit(key) {
    // @ts-ignore
    return types_1.KeyModifierValues[key] || 0;
}
exports.getModifiersBit = getModifiersBit;
function calculateMouseButtonValue(options) {
    if (options.button)
        return options.button;
    return 'left';
}
exports.calculateMouseButtonValue = calculateMouseButtonValue;
